### PR TITLE
Feature/refactor results with mocks

### PR DESCRIPTION
**Pull request by tanyushanel**

### CODE REVIEW

File: exadel-trainee-app/src/app/components/user-results-table/user-results-table.component.ts [hunk 1]
```diff
@@ -18,13 +18,13 @@ import { MatTableDataSource } from '@angular/material/table';
   ],
 })
 export class UserResultsTableComponent implements OnInit, AfterViewInit {
-  results: Test[] = [];
+  MOCK_TEST_RESULTS: TestResult[] = [];
```
Comment (RomPavel) [inline exadel-trainee-app/src/app/components/user-results-table/user-results-table.component.ts:21]: this rename is unnecessary
MOCK_DATA naming is only for variables that contain mock data itself
component variables should be named normal from the begging

File: exadel-trainee-app/src/app/models/dialog-module/user-results-dialog/user-results-dialog.component.ts [hunk 2]
```diff
@@ -23,15 +23,17 @@ export class UserResultsDialogComponent implements OnInit {
...
 
+  length = MOCK_TEST_RESULTS.length;
+
   constructor(
     public dialogRef: MatDialogRef<UserResultsDialogComponent>,
     @Inject(MAT_DIALOG_DATA) public data: User,
   ) {}
 
-  results: Test[] = [];
+  MOCK_TEST_RESULTS: TestResult[] = [];
```
Comment (RomPavel) [inline exadel-trainee-app/src/app/models/dialog-module/user-results-dialog/user-results-dialog.component.ts:33]: same here

File: exadel-trainee-app/src/app/models/user-profile/user-profile.component.ts [hunk 3]
```diff
@@ -1,26 +1,28 @@
...
+import { MOCK_TEST_RESULTS } from '../../../constants/mock-test-results';
 
 @Component({
   selector: 'app-user-profile',
   templateUrl: './user-profile.component.html',
   styleUrls: ['./user-profile.component.scss'],
 })
 export class UserProfileComponent implements OnInit {
-  results: Test[] = [];
+  MOCK_TEST_RESULTS: TestResult[] = [];
```
Comment (RomPavel) [inline exadel-trainee-app/src/app/models/user-profile/user-profile.component.ts:14]: same here

File: exadel-trainee-app/src/app/models/dialog-module/user-results-dialog/user-results-dialog.component.ts [hunk 4]
```diff
@@ -23,15 +23,18 @@ export class UserResultsDialogComponent implements OnInit {
 
   isClicked = false;
 
+  results: Test[] = [];
+
+  length = 0;
```
Comment (RomPavel) [inline exadel-trainee-app/src/app/models/dialog-module/user-results-dialog/user-results-dialog.component.ts:28]: this property can be computed, like 
```
get testsCount() {
       return this.results.length;
}

*ngIf="testsCount"
```

File: exadel-trainee-app/src/constants/mock-test-results.ts [hunk 5]
```diff
@@ -1,240 +1,50 @@
...
+  const toTime = to.getTime();
+  return new Date(fromTime + Math.random() * (toTime - fromTime));
+}
+
+const generateRandomDate = (): Date => {
+  return getRandomDate(new Date('2020-10-12'), new Date('2021-07-12'));
+};
+
+export function generateRandomMark() {
+  return Math.floor(10 + Math.random() * (25 + 1 - 10));
```
Comment (RomPavel) [inline exadel-trainee-app/src/constants/mock-test-results.ts:18]: Just interesting, why you need 25 + 1 - 10

Comment (tanyushanel) [reply exadel-trainee-app/src/constants/mock-test-results.ts]: [from 10 to 25]  Math.floor(min + Math.random() * (max + 1 - min))  just obviuos! Separation creates magic)

File: src/app/models/dialog-module/user-results-dialog/user-results-dialog.component.ts [hunk 6]
```diff
@@ -23,15 +23,21 @@ export class UserResultsDialogComponent implements OnInit {
 
   isClicked = false;
 
+  results: Test[] = [];
+
+  pageSizeOptions = [5];
```
Comment (RomPavel) [inline src/app/models/dialog-module/user-results-dialog/user-results-dialog.component.ts:28]: you dont use it

File: src/app/models/user-profile/user-profile.component.ts [hunk 7]
```diff
@@ -17,10 +17,16 @@ export class UserProfileComponent implements OnInit {
 
   selectedLevel: Level | undefined;
 
+  length = 0;
```
Comment (RomPavel) [inline src/app/models/user-profile/user-profile.component.ts:20]: you dont need this

**Review by RomPavel — APPROVED** [review]

File: exadel-trainee-app/src/constants/mock-test-results.ts (hunk 5 above)
Comment (RomPavel) [reply exadel-trainee-app/src/constants/mock-test-results.ts]: yup, but I guess when people see just
`Math.floor(10 + Math.random() * 16);` 
its obvious that this function return numbers from 10 to 25, and you have less operations and less variables
but nevermind, this additional operations are not really time consuming

File: src/app/models/dialog-module/user-results-dialog/user-results-dialog.component.ts (hunk 6 above)
Comment (tanyushanel) [reply src/app/models/dialog-module/user-results-dialog/user-results-dialog.component.ts:28]: Yep, forgot to take off

**Pull request merged**